### ISSUE TITLE
🐛[BUG]: Model wrapper gaps and limitations

### ISSUE THREAD

**Issue by zubatyuk** [NVIDIA/nvalchemi-toolkit#22]
### Version

0.0

## Differentiation

1. **Missing cross-term differentiation in composite potentials.** `ComposableModelWrapper` calls each child independently and sums precomputed forces numerically (F1 + F2). It never builds a single total energy and differentiates it. When an MLIP produces charges q(R) that feed into a Coulomb potential E_c(q, R), the correct force includes the indirect term (dE_c/dq)(dq/dR). The current code drops this term because each child's forces are computed in isolation. Additionally, MACE clones positions before enabling gradients, so its autograd graph is disconnected from the positions tensor that Coulomb wrappers see.

2. **Blocked two-autograd composition.** Composing two models that both use `forces_via_autograd=True` raises `NotImplementedError`. The guard explicitly rejects this case, noting "energy-first composition (sum energies, single autograd pass) is required for memory correctness but not yet implemented."

## Schema and contracts

3. **Parallel boolean flags.** `ModelConfig` has `compute_forces`, `compute_stresses`, `compute_charges`, etc. `ModelCard` has corresponding `supports_forces`, `supports_stresses`, etc. These parallel lists of booleans must be kept in sync by convention. `_verify_request` checks each pair and only warns on mismatch instead of raising, allowing unsupported computations to proceed silently.

4. **No energy-derived vs direct output distinction.** `ModelCard` has a single boolean `forces_via_autograd` but no concept of which outputs are energy-derived (need autograd) vs. produced directly by the kernel. Without this, the composite cannot correctly decide which children's energies contribute to the autograd graph and which return direct force tensors.

5. **Incomplete composite ModelCard synthesis.** `_build_model_card` does not propagate fields like `needs_node_charges` from sub-models. The synthesized card keeps `needs_node_charges=False` even when a sub-model (e.g. PME) requires charges. Any caller relying on the composite's card to know required batch fields gets incorrect information.

## Batch mutation

6. **adapt_input mutates the caller's batch.** `BaseModelMixin.adapt_input` calls `requires_grad_(True)` directly on the batch's tensors, mutating the caller's data in-place rather than operating on a private copy.

### Minimum reproducible example

```python

```

### Relevant log output

```shell

```

### Environment details

```shell

```
